packages/swapper/src/swappers/thorchain/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,111 @@
+import { fromAssetId, getFeeAssetIdFromAssetId, toChainId } from '@shapeshiftoss/caip'
+import { SupportedChainIds } from '@shapeshiftoss/types'
+
+import { GetTradeQuoteInput, SwapError, SwapErrorTypes, TradeQuote } from '../../../api'
+import { fromBaseUnit } from '../../utils/bignumber'
+import { bnOrZero } from '../../utils/bignumber'
+import { DEFAULT_SLIPPAGE } from '../../utils/constants'
+import { normalizeAmount } from '../../utils/helpers/helpers'
+import { ThorchainSwapperDeps } from '../types'
+import { MAX_THORCHAIN_TRADE } from '../utils/constants'
+import { getThorTxInfo } from '../utils/ethereum/utils/getThorTxData'
+import { getPriceRatio } from '../utils/getPriceRatio/getPriceRatio'
+import { getEthTxFees } from '../utils/txFeeHelpers/ethTxFees/getEthTxFees'
+
+export const getThorTradeQuote = async ({
+  deps,
+  input
+}: {
+  deps: ThorchainSwapperDeps
+  input: GetTradeQuoteInput
+}): Promise<TradeQuote<SupportedChainIds>> => {
+  const { sellAsset, buyAsset, sellAmount, sellAssetAccountId, wallet } = input
+
+  if (!wallet)
+    throw new SwapError('[getTradeQuote] - wallet is required', {
+      code: SwapErrorTypes.VALIDATION_FAILED
+    })
+
+  try {
+    const {
+      assetReference: sellAssetErc20Address,
+      chainReference,
+      chainNamespace
+    } = fromAssetId(sellAsset.assetId)
+
+    const isErc20Trade = sellAssetErc20Address.startsWith('0x')
+    const sellAssetChainId = toChainId({ chainReference, chainNamespace })
```

Comment:
maybe we can take in the appropriate chain adapter as an argument instead of using the manager